nilearn/decomposition/dict_learning.py

```diff
@@ -0,0 +1,187 @@
+"""
+DictLearning
+"""
+
+# Author: Arthur Mensch
+# License: BSD 3 clause
+
+import numpy as np
+from sklearn.externals.joblib import Memory
+from sklearn.linear_model import Ridge
+from sklearn.decomposition import MiniBatchDictionaryLearning
+
+from .._utils import as_ndarray
+from .canica import CanICA
+from .._utils.cache_mixin import CacheMixin
+
+
+class DictLearning(CanICA, MiniBatchDictionaryLearning, CacheMixin):
+    """Perform a map learning algorithm based on component sparsity (rather than independance),
+     over a CanICA initialization, which yields more stable maps than CanICA.
+
+    Parameters
+    ----------
+    mask: Niimg-like object or MultiNiftiMasker instance, optional
+        Mask to be used on data. If an instance of masker is passed,
+        then its mask will be used. If no mask is given,
+        it will be computed automatically by a MultiNiftiMasker with default
+        parameters.
+
+    data: array-like, shape = [[n_samples, n_features], ...]
+        Training vector, where n_samples is the number of samples,
+        n_features is the number of features. There is one vector per
+        subject.
+
+    n_components: int
+        Number of components to extract
+
+    smoothing_fwhm: float, optional
+        If smoothing_fwhm is not None, it gives the size in millimeters of the
+        spatial smoothing to apply to the signal.
+
+    do_cca: boolean, optional
+        Indicate if a Canonical Correlation Analysis must be run after the
+        PCA.
+
+    standardize : boolean, optional
+        If standardize is True, the time-series are centered and normed:
+        their variance is put to 1 in the time dimension.
+
+    threshold: None, 'auto' or float
+        Passed to CanICA. If None, no thresholding is applied. If 'auto',
+        then we apply a thresholding that will keep the n_voxels,
+        more intense voxels across all the maps, n_voxels being the number
+        of voxels in a brain volume. A float value indicates the
+        ratio of voxels to keep (2. means keeping 2 x n_voxels voxels).
```

Comment:
The part in brackets: I think I know what you mean, but it is rather unclear the way it is said here